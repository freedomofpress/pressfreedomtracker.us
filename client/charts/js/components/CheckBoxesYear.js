import React from 'react'

export function CheckBoxesYear({ width, height, options, setSelectedYears, selectedYears }) {
  const side = 24

  return (
    <div style={{ width, display: 'flex', flexDirection: 'column' }}>
      {options.map((d, i) => (
        <div key={d.year}>
          <div
            style={{
              display: 'flex',
              flexDirection: 'row',
              justifyContent: 'space-between',
            }}
          >
            <div
              style={{
                display: 'flex',
                flexDirection: 'row',
                fontSize: 14,
                alignItems: 'center',
                // justifyContent: 'space-between',
                marginBottom: 16,
              }}
            >
              <input
                type="checkbox"
                style={{ width: side, height: side }}
                checked={selectedYears.includes(d.year)}
                onChange={() => {
                  selectedYears.includes(d.year)
                    ? setSelectedYears(selectedYears.filter((y) => y !== d.year))
                    : setSelectedYears([...selectedYears, d.year])
                }}
              />

              <div style={{ flexDirection: 'row', marginLeft: 5, fontFamily: 'sans-serif' }}>
                {d.year}
              </div>
            </div>

            <div className="checkBoxFontFamily" style={{ display: 'flex', fontSize: 12 }}>
              {d.count}
            </div>
          </div>
        </div>
      ))}
    </div>
  )
}
