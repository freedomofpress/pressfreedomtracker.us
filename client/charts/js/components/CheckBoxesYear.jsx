import React from 'react'

export default function CheckBoxesYear({ width, options, selectedYears, onClick }) {
	const side = 20

	return (
		<div style={{ width, display: 'flex', flexDirection: 'column' }}>
			{options.map((d, i) => (
				<div key={d.year}>
					<div
						style={{
							display: 'flex',
							flexDirection: 'row',
							justifyContent: 'space-between',
						}}
					>
						<div
							style={{
								display: 'flex',
								flexDirection: 'row',
								fontSize: 14,
								alignItems: 'center',
								marginBottom: 16,
							}}
						>
							<input
								type="checkbox"
								style={{ width: side, height: side }}
								checked={selectedYears.includes(d.year)}
								onChange={() => onClick(d)}
							/>

							<div style={{ flexDirection: 'row', marginLeft: 10, fontFamily: 'var(--font-base)' }}>
								{d.year}
							</div>
						</div>

						<div
							style={{
								display: 'flex',
								fontSize: 12,
								height: 15,
								fontFamily: 'Helvetica Neue',
								backgroundColor: selectedYears.includes(d.year) ? '#F2FC67' : 'white',
								padding: 3,
							}}
						>
							{d.count}
						</div>
					</div>
				</div>
			))}
		</div>
	)
}
