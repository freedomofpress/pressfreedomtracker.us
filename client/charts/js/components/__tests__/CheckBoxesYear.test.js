import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import CheckBoxesYear from '../CheckBoxesYear'

test('renders CheckBoxesYear with mocked data', () => {
	const renderer = new ShallowRenderer();
	expect(renderer.render(
		<CheckBoxesYear
			width={500}
			options={[{ year: 2020 }]}
			selectedYears={[2020]}
			onClick={() => {}}
		/>
	)).toMatchSnapshot();
});
