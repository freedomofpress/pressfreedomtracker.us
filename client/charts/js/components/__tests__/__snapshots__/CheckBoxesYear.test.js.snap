// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckBoxesYear with mocked data 1`] = `
<div
  style={
    {
      "display": "flex",
      "flexDirection": "column",
      "width": 500,
    }
  }
>
  <div>
    <div
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "fontSize": 14,
            "marginBottom": 16,
          }
        }
      >
        <input
          checked={true}
          onChange={[Function]}
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
          type="checkbox"
        />
        <div
          style={
            {
              "flexDirection": "row",
              "fontFamily": "var(--font-base)",
              "marginLeft": 10,
            }
          }
        >
          2020
        </div>
      </div>
      <div
        style={
          {
            "backgroundColor": "#F2FC67",
            "display": "flex",
            "fontFamily": "Helvetica Neue",
            "fontSize": 12,
            "height": 15,
            "padding": 3,
          }
        }
      />
    </div>
  </div>
</div>
`;
